featuretools/selection/selection.py

```diff
@@ -18,3 +18,102 @@ def remove_low_information_features(feature_matrix, features=None):
+
+    Args:
+        feature_matrix (:class:`pd.DataFrame`): DataFrame whose columns are feature names and rows are instances
+        count_nan_as_value (bool): If True, missing values will be counted as their own unique value.
+                    If set to True, a feature that has one unique value and all other data is missing will be
+                    counted as only having a single unique value. Defaults to False.
+
+    Returns:
+        List of feature names of all the single value features - note None is not counted
+        as a value, so "single value" is either 0 or 1 unique values
+    """
+    unique_counts_by_col = feature_matrix.nunique(dropna=not count_nan_as_value).to_dict()
+    return [f_name for f_name, unique_count in unique_counts_by_col.items() if unique_count <= 1]
+
+
+def find_highly_correlated_features(feature_matrix, pct_corr_threshold=0.95, features_to_check=None, features_to_exclude=None):
+    """
+    Determines whether any pairs of features are highly correlated with one another.
+
+    Args:
+        feature_matrix (:class:`pd.DataFrame`): DataFrame whose columns are feature names and rows are instances
+        pct_corr_threshold (float): The correlation threshold to be considered highly correlated. Defaults to 0.95.
+        cols_to_check (set{str}, optional): Set of column names to check whether any pairs are highly correlated.
+                    If null, defaults to checking all columns.
+        cols_to_exclude set[str], optional): Set of colum names to not check correlation between.
+                    If null, will not exclude any columns.
+
+    Returns:
+        list[tuple(str, str)] of column pairs that is correlated above the set threshold.
+    """
+    if pct_corr_threshold < 0 or pct_corr_threshold > 1:
+        raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
+
+    # --> consider working more with sets if this is gonna be slow
+    if features_to_check is None:
+        features_to_check = feature_matrix.columns
+    else:
+        for f_name in features_to_check:
+            assert f_name in feature_matrix.columns
+
+    if features_to_exclude is not None:
+        features_to_check = [col for col in features_to_check if (col not in features_to_exclude)]
+
+    numeric_dtypes = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
+    boolean = ['bool']
+    numeric_and_boolean_dtypes = numeric_dtypes + boolean
+
+    fm_to_check = (feature_matrix[features_to_check]).select_dtypes(include=numeric_and_boolean_dtypes)
+
+    correlation_by_pairs = {}
+    # Get all pairs of columns and calculate their correlation
+    for f_name1, col1 in fm_to_check.iteritems():
+        for f_name2, col2 in fm_to_check.iteritems():
+            pair = tuple(sorted((f_name1, f_name2)))
+            if f_name1 == f_name2 or pair in correlation_by_pairs:
+                continue
+
+            correlation_by_pairs[pair] = abs(col1.corr(col2))
```

Comment:
we should move the threshold check higher up. that way if we drop a feature early, we dont continue to calculate correlations with it, which is an expensive operation.

Reply:
I definitely agree that we should be more thoughtful here than just "grab every pair," but I'd be worried that this method would keep us from discovering other columns that are correlated to the ones we drop early.

I feel like it'd be really nice to say something like "if `A` and `B` are highly correlated then features built off of them are likely to be highly correlated as well so don't calculate them" but I think that's a different problem than dropping `A` and `B` from further checks.

Another thought I've had is maybe presenting the correlation check as something that should happen on an EntitySet (so basically just the identity features) before running dfs and then using that info to maybe exclude columns from dfs, but that's a whole other thing than running a correlation check after dfs is done.

Reply:
i see your point. it is possible that A/B and B/C are correlated pairs, but A/C aren't. So, if we prematurely drop B, we might miss B/C.

However, i dont think that's a problem. In my mind the goal of this function should be to return a list of features where any pair wont have a correlation about a threshold. There are multiple going to be many sets where that is the case, we just have to return one that is "good".

Reply:
Oh, that's a good way of thinking about it. Would you expect all of these functions to be returning a list of good features  then, or just this one? Up until now, I'd been thinking of it as if we're telling you about which features might be problematic.

Reply:
ya, i think they should all return `(feature_matrix, feature_defs)` like `ft.encode_features`. that was a user can easily call them and continue the process with the good features